gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/CreateApiDomainService.java

```diff
@@ -15,16 +15,143 @@
-    ApiWithFlows create(Api api, AuditInfo auditInfo);
-    Api create(ApiCRD api, AuditInfo auditInfo);
+public class CreateApiDomainService {
+
+    private final ValidateApiDomainService validateApiDomainService;
+    private final ApiCrudService apiCrudService;
+    private final AuditDomainService auditService;
+    private final ApiIndexerDomainService apiIndexerDomainService;
+    private final ApiPrimaryOwnerFactory apiPrimaryOwnerFactory;
+    private final ApiPrimaryOwnerDomainService apiPrimaryOwnerDomainService;
+    private final ApiMetadataDomainService apiMetadataDomainService;
+    private final FlowCrudService flowCrudService;
+    private final NotificationConfigCrudService notificationConfigCrudService;
+    private final ParametersQueryService parametersQueryService;
+    private final WorkflowCrudService workflowCrudService;
+
+    public CreateApiDomainService(
+        ValidateApiDomainService validateApiDomainService,
+        ApiCrudService apiCrudService,
+        AuditDomainService auditService,
+        ApiIndexerDomainService apiIndexerDomainService,
+        ApiMetadataDomainService apiMetadataDomainService,
+        ApiPrimaryOwnerFactory apiPrimaryOwnerFactory,
+        ApiPrimaryOwnerDomainService apiPrimaryOwnerDomainService,
+        FlowCrudService flowCrudService,
+        NotificationConfigCrudService notificationConfigCrudService,
+        ParametersQueryService parametersQueryService,
+        WorkflowCrudService workflowCrudService
+    ) {
+        this.validateApiDomainService = validateApiDomainService;
+        this.apiCrudService = apiCrudService;
+        this.auditService = auditService;
+        this.apiIndexerDomainService = apiIndexerDomainService;
+        this.apiPrimaryOwnerFactory = apiPrimaryOwnerFactory;
+        this.apiPrimaryOwnerDomainService = apiPrimaryOwnerDomainService;
+        this.apiMetadataDomainService = apiMetadataDomainService;
+        this.flowCrudService = flowCrudService;
+        this.notificationConfigCrudService = notificationConfigCrudService;
+        this.parametersQueryService = parametersQueryService;
+        this.workflowCrudService = workflowCrudService;
+    }
+
+    public ApiWithFlows create(Api api, AuditInfo auditInfo) {
+        var primaryOwner = apiPrimaryOwnerFactory.createForNewApi(
+            auditInfo.organizationId(),
+            auditInfo.environmentId(),
+            auditInfo.actor().userId()
+        );
+
+        var sanitized = validateApiDomainService.validateAndSanitizeForCreation(
+            api,
+            primaryOwner,
+            auditInfo.environmentId(),
+            auditInfo.organizationId()
+        );
+
+        if (sanitized.getId() == null) {
+            sanitized.setId(UuidString.generateRandom());
+        }
+
+        var created = apiCrudService.create(
+            sanitized.setEnvironmentId(auditInfo.environmentId()).setCreatedAt(TimeProvider.now()).setUpdatedAt(TimeProvider.now())
```

Comment:
it refers to my remark about `@Accessors` usage
- <personalOpinion> I am not big fan of having regular accessor name chainable, I would prefer the fluent style for this kind of things </personalOpinion>
- what about a `sanitized.createNowForEnvironment(envId)`, my thought here is we can avoid mutating the object directly from the service

Reply:
for readability and as we already set the id earlier, I would keep the "preparation" work before the call instead as passing the chain as parameter of `create` method